documentation/src/main/asciidoc/user_guide/server_protocols_hotrod_wire.adoc

```diff
@@ -2013,6 +2014,151 @@ Here are the current values for each flag: +
 |==============================================================================
 
 
+//TODO? increment version for transaction requests?
+
+
+
+
+.Prepare Request
+
+Request (0x3B):
+
+[cols="3,^2,10",options="header"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Request header
+
+|Xid
+|XID
+|The transaction ID (XID)
+
+|OnePhaseCommit
+|byte
+|When it is set to `1`, the server will use one-phase-commit if available (XA only)
+
+|Number of keys
+|vInt
+|The number of keys
+
+3+^|For each key
+
+| Key Length
+| vInt
+| Length of key.
+Note that the size of a vInt can be up to 5 bytes which in theory can produce bigger numbers than `Integer.MAX_VALUE`.
+However, Java cannot create a single array that’s bigger than `Integer.MAX_VALUE`, hence the protocol is limiting vInt array lengths to `Integer.MAX_VALUE`.
+
+|Key
+|byte array
+|Byte array containing the key
+
+|Version Read
+|long
```

Comment:
what? why complicated? The `RemoteCache` interface uses long (e.g. `removeWithVersion()`), and I don't think we are going to change it. The only issue we have is that the hot rod version is overwriting the entry version (optimistic transactions) and it shouldn't.